batch/aggregator.go

```diff
@@ -94,6 +83,85 @@ func Max(in []schema.Point) float64 {
 	return max
 }
 
+func Mult(in []schema.Point) float64 {
+	valid := false
+	mult := float64(1)
+	for _, term := range in {
+		if math.IsNaN(term.Val) {
+			// NaN * anything equals NaN()
+			mult = math.NaN()
+			break
+		}
+		valid = true
+		mult *= term.Val
+	}
+	if !valid {
+		mult = math.NaN()
+	}
+	return mult
+}
+
+func Med(in []schema.Point) float64 {
+	med := math.NaN()
+	vals := make([]float64, 0, len(in))
+	for i := 0; i < len(in); i++ {
+		p := in[i].Val
+		if !math.IsNaN(p) {
+			vals = append(vals, p)
+		}
+	}
+	if len(vals) != 0 {
+		sort.Float64s(vals)
+		mid := len(vals) / 2
+		if len(vals)%2 == 0 {
+			med = (vals[mid-1] + vals[mid]) / 2
+		} else {
+			med = vals[mid]
+		}
+	}
+	return med
+}
+
+func Diff(in []schema.Point) float64 {
+	diff := math.NaN()
+	for i := 0; i < len(in); i++ {
+		p := in[i].Val
+		if !math.IsNaN(p) {
+			if math.IsNaN(diff) {
+				diff = p
+			} else {
+				diff -= p
+			}
+		}
+	}
+	return diff
+}
+
+func StdDev(in []schema.Point) float64 {
+	avg := Avg(in)
+	if !math.IsNaN(avg) {
+		num := float64(0)
+		totalDeviationSquared := float64(0)
```

Comment:
confusing name. it's not one total deviation that is squared. it's the sum of squares of deviations.